src/main/java/org/jenkinsci/plugins/credentialsbinding/impl/BindingStep.java

```diff
@@ -146,9 +148,9 @@
             final Pattern p = Pattern.compile(pattern.getPlainText());
             return new LineTransformationOutputStream() {
                 @Override protected void eol(byte[] b, int len) throws IOException {
-                    Matcher m = p.matcher(new String(b, 0, len));
+                    Matcher m = p.matcher(new String(b, 0, len, Charsets.UTF_8));
```

Comment:
:bug: this should probably be using `Run.getCharset`. (As noted in https://github.com/jenkinsci/jenkins/pull/2302, it is hard to test these things realistically.)